presto-main/src/main/java/io/prestosql/spiller/AesSpillCipher.java

```diff
@@ -0,0 +1,172 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.spiller;
+
+import io.prestosql.spi.PrestoException;
+
+import javax.crypto.Cipher;
+import javax.crypto.KeyGenerator;
+import javax.crypto.SecretKey;
+import javax.crypto.spec.IvParameterSpec;
+
+import java.security.GeneralSecurityException;
+import java.security.NoSuchAlgorithmException;
+
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+import static io.prestosql.spi.StandardErrorCode.GENERIC_INTERNAL_ERROR;
+
+final class AesSpillCipher
+        implements SpillCipher
+{
+    //  256-bit AES CBC mode
+    private static final String CIPHER_NAME = "AES/CBC/PKCS5Padding";
+    private static final int KEY_BITS = 256;
+
+    private SecretKey key;
+    //  Instance used only for determining encrypted output lengths
+    private Cipher encryptSizer;
```

Comment:
Is there any reason to keep this around after the constructor?  It seems like it is only used for `throwCipherClosedIfNull(encryptSizer)`

Reply:
Yeah, the `encryptSizer` is necessary because of the `encryptedMaxLength(int inputLength)` method. Creating a new encrypt cipher just to determine the output size is slow-ish and expensive since we know the immediate next operation is going to be an encrypt which will create a new cipher instance. Keeping that cipher around (until the instance is closed) to do sizing operations seemed preferable to duplicating the internal logic of the AES engine sizing calculation or creating a fresh instance just to compute the required buffer length.

Reply:
Oh missed `encryptSizer` sizer there due to the chained assignment.

The code might be easier to read if you explicity checked for close in each public method and then just used the fields directly:
```
Failures.checkCondition(key != null, GENERIC_INTERNAL_ERROR, "Spill cipher already closed")); // with a static import
```
I believe it is safe to use the fields directly because this class isn't thread safe anyway.